pyroSAR/drivers.py

```diff
@@ -1173,6 +1173,173 @@ def getCorners(self):
+        facter_obj.close()
+        return facter_m
+    
+    @property
+    def header_filename(self):
+        return self.findfiles(self.pattern)[0]
+    
+    def scanMetadata(self):
+        ################################################################################################################
+        # read header (HDR) file
+        header = self._getHeaderfileContent()
+        header = [head.replace(" ","") for head in header]
+        
+        # read summary text file
+        facter_m = self._parseFacter_m()
+        facter_m = [fact.replace(" ","") for fact in facter_m]
+
+        meta = {}
+        
+        # read polarizations from image file names
+        meta['polarizations'] = [re.search('[HV]{2}', os.path.basename(x)).group(0) for x in self.findfiles('^sar.')]
+        meta['product'] = header[3]
+        ################################################################################################################
+        # read start and stop time --> TODO: in what format is the start and stop time?
+
+        try:
+            start_time = facter_m[168].split('.')[0].zfill(2)+facter_m[168].split('.')[1][:4]
+            stop_time = facter_m[170].split('.')[0].zfill(2)+facter_m[170].split('.')[1][:4]
+        except (AttributeError):
+            raise IndexError('start and stop time stamps cannot be extracted; see file facter_m.dat')
+
+        meta['start'] = str(header[6])#+'T'+start_time
+        meta['stop'] = str(header[6])#+'T'+stop_time
+        ################################################################################################################
+        # read file metadata
+        meta['sensor'] = header[2]
+        ################################################################################################################
+        # read leader file name information
+        meta['acquisition_mode']=header[12]
+        # ###############################################################################################################
+        # read map projection data 
+
+        lat = list(map(float, [header[33], header[35],header[37],header[39]]))
+        lon = list(map(float, [header[34], header[36],header[38],header[40]]))
+
+        meta['corners'] = {'xmin': min(lon), 'xmax': max(lon), 'ymin': min(lat), 'ymax': max(lat)}
+        # print(meta['corners'])
+        # epsg=header[]
+        meta['projection'] = crsConvert(4918, 'wkt') #EPSG: 4918: ITRF97, GRS80
+        ################################################################################################################
+        # read data set summary record
+
+        orbitsPerCycle = 207
+        
+        meta['orbitNumber_abs'] = header[7]
+        meta['orbitNumber_rel'] = header[7]
+        meta['cycleNumber'] = header[5]
+        meta['frameNumber'] = ''
+        
+        meta['lines'] = int(float(facter_m[51]))
+        meta['samples'] = int(float(facter_m[50]))
+        meta['incidence'] = facter_m[119]
```

Comment:
If appropriate, please convert values to `float`:

- incidence
- spacing
- k_dB

or `int`

- cycleNumber
- orbitNumber_rel
- orbitNumber_abs